v1/transform.go

```diff
@@ -0,0 +1,344 @@
+		}
+	}
+
+	return commands
+}
+
+func toV1Operation(op string, resourceOperations []models.ResourceOperation) []v1Models.ResourceOperation {
+	operations := make([]v1Models.ResourceOperation, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		valueMappings := ro.Mappings
+		if op == ResourceOperationSet && len(ro.Mappings) != 0 {
+			valueMappings = reverseMapKeyValue(ro.Mappings)
+		}
+		operations[i] = v1Models.ResourceOperation{
+			Index:          strconv.Itoa(i),
+			Operation:      op,
+			DeviceResource: ro.DeviceResource,
+			Mappings:       valueMappings,
+		}
+	}
+	return operations
+}
+
+func reverseMapKeyValue(mappings map[string]string) map[string]string {
+	valueMappings := make(map[string]string, len(mappings))
+	for k, v := range mappings {
+		valueMappings[v] = k
+	}
+	return valueMappings
+}
+
+func toV1CoreCommand(deviceCommands []models.DeviceCommand) []v1Models.Command {
+	commands := make([]v1Models.Command, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = v1Models.Command{
+			Name: c.Name,
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_R) {
+			commands[i].Get = toV1GetAction(c.Name, c.ResourceOperations)
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_W) {
+			commands[i].Put = toV1PutAction(c.Name, c.ResourceOperations)
+		}
+	}
+
+	return commands
+}
+
+func toV1GetAction(cmdName string, resourceOperations []models.ResourceOperation) v1Models.Get {
+	expectedValues := make([]string, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		expectedValues[i] = ro.DeviceResource
+	}
+	action := v1Models.Action{
+		Path: fmt.Sprintf("/api/v1/device/{deviceId}/%s", cmdName),
+		Responses: []v1Models.Response{
+			{
+				Code:           strconv.Itoa(http.StatusOK),
+				Description:    fmt.Sprintf("Issue the Get command %s", cmdName),
+				ExpectedValues: expectedValues,
+			}, {
+				Code:        strconv.Itoa(http.StatusInternalServerError),
+				Description: "internal server error",
+			},
+		},
+	}
+	return v1Models.Get{Action: action}
+}
+
+func toV1PutAction(cmdName string, resourceOperations []models.ResourceOperation) v1Models.Put {
+	parameterNames := make([]string, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		parameterNames[i] = ro.DeviceResource
+	}
+	action := v1Models.Action{
+		Path: fmt.Sprintf("/api/v1/device/{deviceId}/%s", cmdName),
+		Responses: []v1Models.Response{
+			{
+				Code:        strconv.Itoa(http.StatusNoContent),
+				Description: fmt.Sprintf("Issue the Put command %s", cmdName),
+			}, {
+				Code:        strconv.Itoa(http.StatusInternalServerError),
+				Description: "internal server error",
+			},
+		},
+	}
+	return v1Models.Put{
+		Action:         action,
+		ParameterNames: parameterNames,
+	}
+}
+
+// TransformToV2DeviceProfile transform v1 profile to v2
+func TransformToV2DeviceProfile(profile v1Models.DeviceProfile) models.DeviceProfile {
+	dto := models.DeviceProfile{
```

Comment:
use v2dp as the variable name instead of dto

Reply:
Modified.